web/packages/teleterm/src/ui/ConnectMyComputer/permissions.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Cluster } from 'teleterm/services/tshd/types';
+import { ConfigService } from 'teleterm/services/config';
+import { RuntimeSettings } from 'teleterm/mainProcess/types';
+
+/**
+ * Checks if Connect My Computer can be used for the given root cluster.
+ *
+ * The root cluster is required because `loggedInUser` and `features` are not fully defined for leaves.
+ * */
+export function canUseConnectMyComputer(
```

Comment:
Looking now at the code, I think long term we're definitely going to need something like cancancan where we can have [a single place which defines permissions](https://github.com/CanCanCommunity/cancancan#define-abilities). Otherwise we'll have to jump around the codebase to understand what the user can and cannot do.

The API doesn't have to be the same as that of cancancan, its API is quite extensible and made for dynamically-typed lang, so it'll be hard to match that in TypeScript.

The important thing for us is just to have it all in a single place. So I imagine for a start we could even have a workspace-level context. The context would provide functions such as `canUseConnectMyComputer` which would take no arguments.

Could you add a TODO comment about this to this function?

Reply:
I absolutely agree, having a single place would be much better. AFAIK we don't have permissions checking anywhere else in the app, so I decided for a simple function. But as we want to add permissions to other places, such as Servers/Dbs/Kubes, it will be a perfect opportunity to organize them better. TODO added.